### PR TITLE
add(illumina run metrics table)

### PR DESCRIPTION
**Pull request by ChrOertlin**
## Description
Closes: https://github.com/Clinical-Genomics/add-new-tech/issues/3
### Added

-

### Changed

-

### Fixed

-

### How to prepare for test

- [ ] Ssh to relevant server (depending on type of change)
- [ ] Use stage: `us`
- [ ] Paxa the environment: `paxa`
- [ ] Install on stage (example for Hasta):
    ```shell
    bash /home/proj/production/servers/resources/hasta.scilifelab.se/update-tool-stage.sh -e S_cg -t cg -b [THIS-BRANCH-NAME] -a
    ```

### How to test

- [ ] Do ...

### Expected test outcome

- [ ] Check that ...
- [ ] Take a screenshot and attach or copy/paste the output.

## Review

- [ ] Tests executed by
- [ ] "Merge and deploy" approved by
Thanks for filling in who performed the code review and the test!

### This [version](https://semver.org/) is a

- [ ] **MAJOR** - when you make incompatible API changes
- [ ] **MINOR** - when you add functionality in a backwards compatible manner
- [ ] **PATCH** - when you make backwards compatible bug fixes or documentation/instructions

## Implementation Plan

- [ ] Document in ...
- [ ] Deploy this branch on ...
- [ ] Inform to ...

### CODE REVIEW

**Review by seallard — APPROVED** [review]
Looks good 👍

File: alembic/versions/2024_05_16_6e6c36d5157b_add_illumina_run_metrics.py [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+branch_labels = None
+depends_on = None
+
+
+def upgrade():
+    op.create_table(
+        "illumina_run_metrics",
+        sa.Column("id", sa.ForeignKey("run_metrics.id"), nullable=False, primary_key=True),
+        sa.Column("sequencer_type", sa.String(length=32), nullable=True),
+        sa.Column("sequencer_name", sa.String(length=32), nullable=True),
```
[the remark this answers is not in the document]
Comment (diitaz93) [reply alembic/versions/2024_05_16_6e6c36d5157b_add_illumina_run_metrics.py:27]: This was decided upon in the refinement, we are following this inheritance model https://miro.com/app/board/uXjVN-I19i8=/

[the remark this answers is not in the document]
Comment (ChrOertlin) [reply alembic/versions/2024_05_16_6e6c36d5157b_add_illumina_run_metrics.py:27]: I think so, yes. Will bring this up in a rfinement around this project (probably tomorrow)

File: cg/store/models.py [hunk 2]
```diff
@@ -989,6 +989,37 @@ class RunMetrics(Base):
...
+    total_yield: Mapped[BigInt | None]
+    yield_q30: Mapped[Num_6_2 | None]
+    cycles: Mapped[int]
+    demultiplexing_software: Mapped[Str32 | None]
+    demultiplexing_version: Mapped[Str32 | None]
+    sequencing_started_at: Mapped[datetime | None]
+    sequencing_completed_at: Mapped[datetime | None]
+    demultiplexing_started_at: Mapped[datetime | None]
+    demultiplexing_completed_at: Mapped[datetime | None]
+
```
[the remark this answers is not in the document]
Comment (ChrOertlin) [reply cg/store/models.py:1019]: the relationship should be on the parent

File: alembic/versions/2024_05_16_6e6c36d5157b_add_illumina_run_metrics.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (ChrOertlin) [reply alembic/versions/2024_05_16_6e6c36d5157b_add_illumina_run_metrics.py:27]: @diitaz93 I think what @seallard  meant is to have an extra table that has all the illumina sequencers with the type, name as information- we can that link these tables. That way we don't need to repeatedly store type and name, but only have a foreign key reference to that table. I think that makes sense.

[the remark this answers is not in the document]
Comment (diitaz93) [reply alembic/versions/2024_05_16_6e6c36d5157b_add_illumina_run_metrics.py:27]: Ahh ok yes I understand now

**Review by Vince-janv — APPROVED** [review]
Well done 💪

**Comment by ChrOertlin** [comment]
image

**Pull request merged**